src/utils/platform.ts

```diff
@@ -1,56 +1,38 @@
-export function isChromiumBased() {
-    return navigator.userAgent.toLowerCase().includes('chrome') || navigator.userAgent.toLowerCase().includes('chromium');
-}
-
-export function isFirefox() {
-    return navigator.userAgent.includes('Firefox');
-}
-
-export function isVivaldi() {
-    return navigator.userAgent.toLowerCase().includes('vivaldi');
-}
-
-export function isYaBrowser() {
-    return navigator.userAgent.toLowerCase().includes('yabrowser');
-}
-
-export function isOpera() {
-    const agent = navigator.userAgent.toLowerCase();
-    return agent.includes('opr') || agent.includes('opera');
-}
-
-export function isEdge() {
-    return navigator.userAgent.includes('Edg');
-}
-
-export function isWindows() {
-    if (typeof navigator === 'undefined') {
-        return null;
-    }
-    return navigator.platform.toLowerCase().startsWith('win');
-}
-
-export function isMacOS() {
-    if (typeof navigator === 'undefined') {
-        return null;
+import {PlatformData} from '../definitions';
+
+export const platformData: PlatformData = {} as any;
+
+export function runPlatformTest() {
+    const userAgent = navigator.userAgent.toLowerCase();
+    const platform = navigator.platform.toLowerCase();
+    platformData.isChromium = userAgent.includes('chrome') || userAgent.includes('chromium');
+    platformData.isFirefox = userAgent.includes('firefox');
+    platformData.isVivaldi = userAgent.includes('vivaldi');
+    platformData.isYaBrowser = userAgent.includes('yabrowser');
+    platformData.isOpera = userAgent.includes('opr') || userAgent.includes('opera');
+    platformData.isEdge = userAgent.includes('edg');
+    platformData.isWindows = platform.startsWith('win');
+    platformData.isMacOS = platform.startsWith('mac');
+    platformData.isMobile = userAgent.includes('mobile');
+    const m = userAgent.match(/chrom[e|ium]\/([^ ]+)/);
+    if (m && m[1]) {
+        platformData.chromiumVersion = m[1];
+    } else {
+        platformData.chromiumVersion = '';
     }
-    return navigator.platform.toLowerCase().startsWith('mac');
-}
-
-export function isMobile() {
-    if (typeof navigator === 'undefined') {
-        return null;
+    try {
```

Comment:
Try self-executing function to hide logic out of root scope
```typescript
export const isDefinedSelectorSupported = (() => {
    try {	
        document.querySelector(':defined');	
        return true;	
    } catch (err) {	
        return false;	
    }
})();
```
Note () at the end.
Same for similar cases.